### PR TITLE
make send recv always sync

### PR DESCRIPTION
**Pull request by typhoonzero**
Let this merge to resolve current version hung problems.

### CODE REVIEW

**Comment by Yancey0623** [comment]
Just a note, this PR would reduce the performance by about 12%, https://github.com/PaddlePaddle/Paddle/pull/11515#issuecomment-398266557

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 1]
```diff
@@ -225,6 +225,7 @@ def transpile(self,
                 inputs={"X": splited_vars},
                 outputs={},
                 attrs={
+                    "sync_mode": self.sync_mode,  # convert to int
```
Comment (Yancey0623) [inline python/paddle/fluid/transpiler/distribute_transpiler.py]: Please add a TODO comment, and we need to make it as `async_send` to improve performance.

**Comment by typhoonzero** [comment]
Should re-enable full async feature once we have resolve all possible issues for distributed parallel training.

File: python/paddle/fluid/transpiler/distribute_transpiler.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/fluid/transpiler/distribute_transpiler.py]: Done

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Comment by typhoonzero** [comment]
Closing this PR, as we discussed, `req_count++` is not called in the IO thread, if the order of operator execution is correct, later send_barrier can wait that previouse RPC call utils the end, that means the copy and send is finished.

**Pull request closed without merging**